src/app/reporting/EventLoggingTypes.h

```diff
@@ -0,0 +1,227 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Copyright (c) 2015-2017 Nest Labs, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *
+ * @brief
+ *   Enums, types, and tags used in Chip Event Logging.
+ *
+ */
+
+#pragma once
+
+#include <core/CHIPCore.h>
+#include <core/CHIPTLV.h>
+#include <system/SystemPacketBuffer.h>
+#include <util/basic-types.h>
+
+namespace chip {
+namespace app {
+namespace reporting {
+
+/**
+ * @brief
+ *   The Priority of the log entry.
+ *
+ * @details
+ * Priority is used as a way to filter events before they are
+ * actually emitted into the log. After the event is in the log, we
+ * make no further provisions to expunge it from the log.
+ * The priority level serves to prioritize event storage. If an
+ * event of high priority is added to a full buffer, events are
+ * dropped in order of priority (and age) to accommodate it. As such,
+ * priority levels only have relative value. If a system is
+ * using only one priority level, events are dropped only in order
+ * of age, like a ring buffer.
+ */
+
+typedef enum PriorityLevel
```

Comment:
Can we make this an enum class?  We may need to define a comparison operator for it if we want to do ordinal comparisons, but that's ok...

Reply:
Done, thanks